lms/djangoapps/discussion_api/api.py

```diff
@@ -110,6 +110,42 @@ def _is_user_author_or_privileged(cc_content, context):
     )
 
 
+def _validate_and_update_vote_count(actions_form, serializer, update_data):
+    """
+    Calls is_valid on form, updates vote, then calls is_valid on serializer
+
+    Arguments:
+
+        actions_form (Form): Form with information to update
+        serializer (ThreadSerializer/CommentSerializer): Serializer to update
+        update_data (dict): Dict of the fields to be updated
+    """
+    if not actions_form.is_valid():
+        raise ValidationError(dict(actions_form.errors.items()))
+    if "voted" in update_data:
+        _update_vote_count(actions_form.cleaned_data["voted"], serializer)
+    if not serializer.is_valid():
+        raise ValidationError(dict(serializer.errors.items()))
+
+
+def _update_vote_count(voted, serializer):
+    """
+    Updates the vote_count in the given serializer
+
+    If the user has already voted and `voted == False` then -1 to the vote count
+    If the user has not voted and `voted == True` then +1 to the vote count
+
+    Arguments:
+
+        voted (Bool): Boolean of the new vote status
+        serializer (ThreadSerializer/CommentSerializer): Serializer to update
+    """
+    if voted and not serializer.data["voted"]:
+        serializer.data["vote_count"] += 1
+    elif not voted and serializer.data["voted"]:
+        serializer.data["vote_count"] -= 1
```

Comment:
What happens with this `vote_count`?  Is it passed to Ruby?  Is it returned in the response?
In either case, shouldn't the Ruby code be the real owner of the vote_count?